packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+                        />
+                      </Box>
+                      <FieldInput
+                        mb={2}
+                        label="AWS ARN"
+                        onChange={e =>
+                          setRuleAtIndex(index, {
+                            ...rules[index],
+                            awsArn: e.target.value,
+                          })
+                        }
+                        placeholder="arn:aws:sts::111111111111:assumed-role/teleport-node-role/i-*"
+                        value={rule.awsArn}
+                      />
+                    </>
+                  )}
+                </RuleBox>
+              ))}
+            </Box>
+            <Box
+              onClick={() =>
+                setRules([
+                  ...rules,
+                  { awsAccount: '', awsArn: '', isCollapsed: false },
+                ])
+              }
+            >
+              <ButtonAddRule>+ Add new rule</ButtonAddRule>
+            </Box>
+            {token && (
+              <>
+                <Text mt="3" mb="2" bold typography="h5">
+                  Script Generated
+                </Text>
+                <Text>
+                  Start teleport with the following parameters. The script will
+                  be valid for{' '}
+                  <Text bold as={'span'}>
+                    {expiry}.
+                  </Text>
+                </Text>
+                <Box>
+                  <TextSelectCopy
+                    mt="2"
+                    text={`teleport start --roles=node --token=${token} --auth-server=${host} --join-method=iam`}
+                  />
+                </Box>
+              </>
+            )}
+          </DialogContent>
+          <DialogFooter>
+            <ButtonPrimary
+              mr="3"
+              disabled={attempt.status === 'processing'}
+              onClick={() => handleGenerate(validator)}
+            >
+              {token ? 'Regenerate Script' : 'Generate Script'}
+            </ButtonPrimary>
+            <ButtonSecondary onClick={onClose}>Close</ButtonSecondary>
+          </DialogFooter>
+        </>
+      )}
+    </Validation>
+  );
+}
+
+const RuleBox = styled(Box)`
+  margin: 4px 0 12px 0;
+  position: relative;
+
+  &:last-of-type {
+    margin-bottom: 8px;
+  }
+`;
+
+const ButtonAddRule = styled(ButtonLink)`
+  font-weight: bold;
+  font-size: 14px;
+  color: white;
+  text-decoration: none;
+`;
+
+const ButtonEditRule = styled(ButtonLink)`
+  position: absolute;
+  top: 12px;
+  right: 12px;
+  font-size: 16px;
+  text-decoration: none;
+`;
+
+const ButtonRemoveRule = styled(ButtonLink)`
+  position: absolute;
+  top: 12px;
+  right: 12px;
+`;
+
+export const AWS_ACC_ID_REGEXP = /^\d{12}$/;
+const requiredAwsAccount = (value, rule: Rule) => () => {
+  if (!rule.awsAccount && !rule.awsArn) {
+    return {
+      valid: false,
+      message: 'Rule cannot be empty',
+    };
+  }
```

Comment:
i think we need to re-think how to do this b/c I was initially confused (b/c i'm one of those who doesn't really read error unless i have to :), so i initially thought the first input box had to be required, and then i thought it was a bug when that "requirement" went away when I filled the second box. I would maybe ask rishi the best way to design the `either or, or both` case. Otherwise it might be better to just highlight the whole box as red, and put like `either account ID or ans is required` next to the rule title `Rule #X`

Reply:
I agree, I'll reach Rishi and ask for his help to improve the UX there, thanks

Reply:
Update:

The team agreed on going for a simple flow here, making the AWS Account ID mandatory and arn optional.

If the user wants to do something different, they can read the docs; the UI will aim for solving the simplest use case to get users starting.